### PR TITLE
[tests/checkpointing] refactor with BoringModel

### PR DESCRIPTION
**Pull request by ydcjeff**
## What does this PR do?

Refactor with BoringModel for tests in tests/checkpointing.

Part of #4501

Imports are sorted by isort.

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by rohitgr7** [comment]
quick question... unrelated to this PR. Why do we run cpu tests on drone?? I mean if it's not necessary, wouldn't it reduce the drone runtime a lot? CPU tests already run in different envs.

**Comment by edenlightning** [comment]
> quick question... unrelated to this PR. Why do we run cpu tests on drone?? I mean if it's not necessary, wouldn't it reduce the drone runtime a lot? CPU tests already run in different envs.

@Borda

**Review by tchaton — APPROVED** [review]
Great work !

File: tests/checkpointing/test_model_checkpoint.py [hunk 1]
```diff
@@ -14,27 +14,39 @@
...
 from pytorch_lightning import Trainer, seed_everything
 from pytorch_lightning.callbacks import ModelCheckpoint
 from pytorch_lightning.loggers import TensorBoardLogger
-from tests.base import EvalModelTemplate, BoringModel
 from pytorch_lightning.utilities.cloud_io import load as pl_load
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.base import BoringModel
+
+
+class LessBoringModel(BoringModel):
```
Comment (Borda) [inline tests/checkpointing/test_model_checkpoint.py:40]: as this is not inside a test function, can we have more description name?

File: tests/checkpointing/test_model_checkpoint.py [hunk 2]
```diff
@@ -462,12 +472,43 @@ def test_default_checkpoint_behavior(tmpdir):
 
 
 def test_ckpt_metric_names_results(tmpdir):
-    model = EvalModelTemplate()
-    model.training_step = model.training_step_result_obj
+    class LessBoringModel(BoringModel):
```
Comment (Borda) [inline tests/checkpointing/test_model_checkpoint.py:475]: this is a bit confusing again the same model name

**Review by williamFalcon — APPROVED** [review]

File: tests/checkpointing/test_model_checkpoint.py (hunk 1 above)
Comment (ydcjeff) [reply tests/checkpointing/test_model_checkpoint.py]: Done

File: tests/checkpointing/test_model_checkpoint.py (hunk 2 above)
Comment (ydcjeff) [reply tests/checkpointing/test_model_checkpoint.py]: Changed

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by rohitgr7 — APPROVED** [review]

**Review by zech564 — APPROVED** [review]

**Pull request merged**